src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs

```diff
@@ -993,5 +999,54 @@ private void AppendBoundOperationOnNavigationPropertyPath(IEdmOperation edmOpera
                 }
             }
         }
+
+        private void AppendBoundOperationOnOperationPath(IEdmOperation edmOperation, bool isCollection, IEdmEntityType bindingEntityType)
+        {
+            bool isEscapedFunction = _model.IsUrlEscapeFunction(edmOperation);
+
+            // only composable functions
+            var paths = _allOperationPaths.Where(x => x.LastSegment is ODataOperationSegment operationSegment
+                                                    && operationSegment.Operation is IEdmFunction edmFunction
+                                                    && edmFunction.IsComposable).ToList();
```

Comment:
```suggestion
                                                    && edmFunction.IsComposable);
```

Reply:
`AppendPath` inside the `foreach` adds a new path to `_allOperationPaths`.

Reply:
yes, but in fact, `_allOperationPaths` will be enumerated and an attempt to add a new operation path to `_allOperationPaths` will throw `InvalidOperationException : Collection was modified; enumeration operation may not execute.`